sensor/common/deduper/deduper.go

```diff
@@ -15,16 +15,61 @@ var (
 	log = logging.LoggerForModule()
 )
 
-// key is the key by which messages are deduped.
-type key struct {
-	id           string
-	resourceType reflect.Type
+// Key is the key by which messages are deduped.
+type Key struct {
+	ID           string
+	ResourceType reflect.Type
 }
```

Comment:
Instead of making the fields public, I would use the interface:

https://github.com/stackrox/stackrox/blob/8350687d206a604acd7898517088cde33d587319/sensor/common/deduper/deduper.go#L45-L48